airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py

```diff
@@ -501,6 +508,16 @@ def _validate_and_transform_start_date(self, start_date: str) -> datetime.date:
 
         return start_date
 
+    def _validate_and_transform_end_date(self, end_date: str) -> datetime.date:
```

Comment:
Why not refactor the previous function and use for both start and end date?
```suggestion
    def _validate_and_transform_date(self, date: str, default_date) -> datetime.date:
        date = default_date if not date else date
        try:
            date = utils.string_to_date(date)
        except ValueError as e:
            raise ConfigurationError(str(e))

        return date
```

Then call the function as:
```python
        config["date_ranges_end_date"] = self._validate_and_transform_date(
date=config.get("date_ranges_end_date"), default_date=self.default_date_ranges_end_date)

```

Reply:
I like it, good call - I added your changes to the MR - thank you! 👍